rasgotransforms/rasgotransforms/transforms/plot/plot.sql

```diff
@@ -0,0 +1,99 @@
+{%- set axis_type_dict = get_columns(source_table) -%}
+{%- set axis_type_response = axis_type_dict[x_axis].upper() -%}
+{%- if 'DATE' in axis_type_response or 'TIME' in axis_type_response -%}
+  {%- set axis_type = "date" -%}
+{%- elif 'NUM' in axis_type_response or 'FLOAT' in axis_type_response or 'INT' in axis_type_response or 'DECIMAL' in axis_type_response or 'DOUBLE' in axis_type_response or 'REAL' in axis_type_response -%}
+    {%- set axis_type = "numeric" -%}
+{%- elif 'BINARY' in axis_type_response or 'TEXT' in axis_type_response or 'BOOLEAN' in axis_type_response or 'CHAR' in axis_type_response or 'STRING' in axis_type_response or 'VARBINARY' in axis_type_response -%}
+    {%- set axis_type = "categorical" -%}
+{%- else -%}
+    {{ raise_exception('The column selected as an axis is not categorical, numeric, or datetime. Please choose an axis that is any of these data types and recreate the transform.') }}
+{%- endif -%}
+
+{%- if num_buckets is not defined -%}
+    {%- set bucket_count = 200 -%}
+{%- else -%}
+    {%- set bucket_count = num_buckets -%}
+{%- endif -%}
+
+-- if the axis is continuous or a date, do a line chart
+{% if axis_type in ['date', 'numeric'] -%}
+    WITH AXIS_RANGE AS (
+    -- Use a user-defined axis column to calculate the min & max of the axis (and buckets on the axis)
+    SELECT
+        {% if axis_type == 'date' -%}
+            MIN(DATE_PART(EPOCH_SECOND, {{ x_axis }}))-1 AS MIN_VAL
```

Comment:
Probably a dumb question a few weeks late: why cast to epoch here? Is this what the UI prefers and we punt timestamp formatting to it?

Reply:
We're actually not sending them as epoch time in the query response - the UI is getting datetimes. The cast to epoch is necessary for some of the math that we do later in the query